### PR TITLE
Torch 1.3

### PR DESCRIPTION
**Pull request by PonteIneptique**
Hey !
I spent some time looking at what could be blocking >= 1.2. Most of it was the use of int mask used as boolean, which was touched by a breaking change inserted in 1.2.0 ( https://github.com/pytorch/pytorch/releases/tag/v1.2.0 )
I found two concerned occurrences, trained with my usual models as well as tagged. This does not cover all models but gives a general robustness.

All other masks seems to use the numerical value rather than the boolean one.

### CODE REVIEW

File: pie/models/attention.py [hunk 1]
```diff
@@ -163,7 +163,8 @@ def forward(self, dec_out, enc_outs, lengths):
         # (batch x src_seq_len) => (trg_seq_len x batch x src_seq_len)
         mask = mask.unsqueeze(0).expand_as(weights)
         # weights = weights * mask.float()
-        weights.masked_fill_(1 - mask, -float('inf'))
+        # Torch 1.1 -> 1.2: (1 - mask) becomes ~(mask)
+        weights.masked_fill_(~(mask), -float('inf'))
```
Comment (emanjavacas) [inline pie/models/attention.py]: I think you can just use `~mask~` (no need for parens)

Comment (PonteIneptique) [reply pie/models/attention.py]: I'll see if this works. I'll probably bump the version if you are okay to make a release :) Last torch allows for easier use of current CUDNN :)

File: pie/torch_utils.py [hunk 2]
```diff
@@ -228,11 +228,14 @@ def word_dropout(inp, p, training, encoder):
...
     ).float().to(inp.device)
     # compute bernoulli mask
-    mask = 1 - torch.bernoulli(mask / (p + mask))
+    # PyTorch 1.2 and 1.3 :
+    #   - Deprecated masked_fill_ used with int.
+    #   - Mask should be booleans in general
+    mask = ((1 - torch.bernoulli(mask / (p + mask))) == 1)
     # don't drop padding
-    mask.masked_fill_(inp.eq(encoder.get_pad()), 0)
+    mask.masked_fill_((inp.eq(encoder.get_pad()) == 1), 0)
```
Comment (emanjavacas) [inline pie/torch_utils.py]: Same as before.

File: pie/torch_utils.py [hunk 3]
```diff
@@ -228,11 +228,14 @@ def word_dropout(inp, p, training, encoder):
           for j in range(batch)] for i in range(seq_len)]
     ).float().to(inp.device)
     # compute bernoulli mask
-    mask = 1 - torch.bernoulli(mask / (p + mask))
+    # PyTorch 1.2 and 1.3 :
+    #   - Deprecated masked_fill_ used with int.
+    #   - Mask should be booleans in general
+    mask = ((1 - torch.bernoulli(mask / (p + mask))) == 1)
```
Comment (emanjavacas) [inline pie/torch_utils.py]: I find this a bit hard to read. Couldn't we just do:

```
mask = 1 - torch.bernoulli(mask / (p + mask))
mask = mask.byte()
```

**Comment by emanjavacas** [comment]
Hey thanks for the PR! I haven't been using pytorch > 1.0 and I am not up to date with these changes. Please have a look at the comment (mostly about readability).

File: pie/torch_utils.py (hunk 3 above)
Comment (PonteIneptique) [reply pie/torch_utils.py]: mask.byte() is actually the issue, should have used bool but did not see it in the doc...

**Comment by PonteIneptique** [comment]
Here you go ! Thanks for the quick answer :)

**Comment by emanjavacas** [comment]
Thanks for the PR!

**Comment by PonteIneptique** [comment]
Thanks for the quick management ;) When you have time, a bump to "0.2.6" would be neat ;)
